testing/tests/DevExpress.ui.widgets/overlay.tests.js

```diff
@@ -780,31 +780,156 @@ testModule('visibility', moduleConfig, () => {
         }
     });
 
-    test('overlay should not be shown if e.cancel == true in the onShowing event handler (T825865)', function(assert) {
-        const onShown = sinon.stub();
-        const $overlay = $('#overlay').dxOverlay({
-            onShowing: e => e.cancel = true,
-            onShown
+    testModule('e.cancel', {
+        beforeEach: function(assert) {
+            this.onShown = sinon.stub();
+            this.onHidden = sinon.stub();
+            this.$overlay = $('#overlay').dxOverlay({
+                onShown: this.onShown,
+                onHidden: this.onHidden,
+            });
+            this.overlay = this.$overlay.dxOverlay('instance');
+            this.checkIsNotShown = () => {
+                assert.notOk(this.onShown.called, 'onShown should not be called');
+            };
+            this.checkIsShown = () => {
+                assert.notOk(this.$overlay.is(':hidden')), 'overlay is not hidden';
+                assert.ok(this.onShown.called, 'onShown should be called');
+            };
+            this.checkIsNotHidden = () => {
+                assert.notOk(this.$overlay.is(':hidden')), 'overlay is not hidden';
+                assert.notOk(this.onHidden.called, 'onHidden should not be called');
+            };
+            this.checkIsHidden = () => {
+                assert.ok(this.$overlay.is(':hidden')), 'overlay is hidden';
+                assert.ok(this.onHidden.called, 'onHidden should be called');
+            };
+        }
+    }, () => {
+        test('overlay should not be shown if e.cancel == true in the onShowing event handler (T825865)', function(assert) {
+            this.overlay.option({ onShowing: e => e.cancel = true });
+            this.overlay.show();
+
+            assert.ok(this.$overlay.is(':hidden')), 'overlay is hidden';
```

Comment:
Here's a misstype with brackets.

Besides, why this assert is not inside `checkIsNotShown` function how it's done for all another similar functions?

Reply:
because we actualy show the Overlay (but hide it via `visibility: hidden`) before we resolve cancel